[core] Use PicoCli and unify PMD usage under a single main (#4059)
pmd/pmd: docs/pages/next_major_development.md

The new logo is available from the [Logo Project Page](pmd_projectdocs_logo.html).

### Revamped Command Line Interface

> PMD now ships with a unified Command Line Interface for both Linux/Unix and Windows. Under the `pmd` executable (`pmd.bat` for Windows),
> all utilities can be launched similar to how git is normally used. All commands and options are thoroughly documented in the help,

Reviewer: I think this reference to git comes a bit out of nowhere. I'd suggest to rephrase to make it clear what the delta is

```suggestion
PMD now ships with a unified Command Line Interface for both Linux/Unix and Windows. Instead of having a collection of scripts 
for the different utilities shipped with PMD, a single script `pmd` (`pmd.bat` for Windows) can now launch all
utilities using subcommands, e.g. `pmd run`, `pmd designer`. All commands and options are thoroughly documented in the help,
```

with full color support where available. Moreover, efforts were made to provide consistency in the usage of all PMD utilities.

```
$ pmd --help    
Usage: pmd [-hV] [COMMAND]
  -h, --help      Show this help message and exit.
  -V, --version   Print version information and exit.
Commands:
  analyze, analyse, run  The PMD standard source code analyzer
  cpd                    Copy/Paste Detector - find duplicate code
  designer               The PMD visual rule designer
  cpd-gui                GUI for the Copy/Paste Detector
                           Warning: May not support the full CPD feature set
  ast-dump               Experimental: dumps the AST of parsing source code
Exit Codes:

>   0   Succesful analysis, no violations found

Reviewer: ```suggestion
  0   Successful analysis, no violations found
```

  1   An unexpected error occurred during execution
  2   Usage error, please refer to the command help
  4   Successful analysis, at least 1 violation found
```

> 

Reviewer: I would add an example here:
````suggestion

For instance, where you previously would have run
```shell
run.sh pmd -d src -R ruleset.xml
```
you should now use
```shell
pmd run -d src -R ruleset.xml
```
````

The new logo is available from the [Logo Project Page](pmd_projectdocs_logo.html).

> ### Revamped Command Line Interface

Author: I hadn't noticed that `7_0_0_release_notes.md`… I'm a little confused as to why we have these separate files. What purpose does each have? should we merge them and keep only one around?

> ### Revamped Command Line Interface

Reviewer 2: As far as I remember: `next_major_development.md` is used to communicate to the community, what we are going to do in PMD 7 (the next major version) - like a preview or "what you can expect". Thus it is in master and should describe the current progress. It is available in all the PMD 6 versions that we release: https://pmd.github.io/pmd-6.50.0/pmd_next_major_development.html

`7_0_0_release_notes.md` are the release notes for PMD 7. We renamed this file from `release_notes.md`, so that we don't overwrite it always when we merge from master->pmd/7.0.x. That's why it is a different file.

For this PR, I'd suggest to move just this new section into `7_0_0_release_notes.md` and leave `next_major_development.md` untouched. May we should remove this file in the pmd/7.0.x branch - since I expect it to be *not* included in the final PMD 7 release.
The only important part are the API changes, that we probably will need to put into the final PMD 7 release notes when the time comes.

PMD now ships with a unified Command Line Interface for both Linux/Unix and Windows. Instead of having a collection of scripts 
for the different utilities shipped with PMD, a single script `pmd` (`pmd.bat` for Windows) can now launch all
utilities using subcommands, e.g. `pmd run`, `pmd designer`. All commands and options are thoroughly documented in the help,
with full color support where available. Moreover, efforts were made to provide consistency in the usage of all PMD utilities.

```
$ pmd --help    
Usage: pmd [-hV] [COMMAND]
  -h, --help      Show this help message and exit.
  -V, --version   Print version information and exit.
Commands:

>   analyze, analyse, run  The PMD standard source code analyzer

Reviewer: I'm not sure we should alias this command. `analyze` is long and has spelling variants, so maybe we should avoid it. `run` is a bit nondescript... Maybe `pmd check` would be appropriate? In Rust you call `cargo check` to check your code

Reviewer 2: :+1: I like `check`.

Author: Agreed, changing it!
